### PR TITLE
[cli] Fix installing patch range lib defs

### LINKED ISSUE #3991: Installing koa@2.13.0 keeps resolving to koa@2.0.x instead of koa@2.x.x
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
With koa, there are defs for `2.0.x` and `2.x.x`, when I try to install koa defs, it keeps resolving to `2.0.x` even though my dependency doesn't satisfy that, I'm on `^2.13.0` for reference.

Given that `2.0.x` -> `2.x.x` is a minor and non-breaking change, can we remove `koa@2.0.x` lib defs?  Which fixes my issue but ideally flow-typed cli should just give me the right def so that I don't have this issue.

**If the current behavior is a bug, please provide the steps to reproduce.**

Have latest koa install in package.json
```
service-yangtze git:(master) ✗ flow-typed install koa
• Searching for 1 libdefs...
• rebasing flow-typed cache...
• Installing 1 libDefs...
  • koa_v2.0.x.js
    └> ./flow-typed/npm/koa_v2.0.x.js
koa
/Users/brianchen/projects/service-yangtze/flow-typed/npm
• The following installed libdefs are compatible with your dependencies, but may not include all minor and patch changes for your specific dependency version:

  • libdef: koa_v2.0.x (satisfies koa@^2.13.0)

  Consider submitting a versioned update for this package to
  https://github.com/flowtype/flow-typed/
```

**What is the expected behavior?**
It should have installed `koa_v2.x.x` instead

**Local Environment Information**
If this issue is specific to the CLI, provide the results of running `npx envinfo` from the command line in a GitHub gist.
```
System:
    OS: macOS 10.15.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 478.41 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.17.0 - ~/.nvm/versions/node/v12.17.0/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.17.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    Cargo: 1.46.0 - ~/.cargo/bin/cargo
    Homebrew: 2.5.7 - /usr/local/bin/brew
    pip3: 20.1.1 - /usr/local/bin/pip3
    RubyGems: 3.0.3 - /usr/bin/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 4.2.1 - /usr/bin/gcc
    Git: 2.28.0 - /usr/local/bin/git
    Clang: 1103.0.32.62 - /usr/bin/clang
  Servers:
    Apache: 2.4.41 - /usr/sbin/apachectl
  SDKs:
    iOS SDK:
      Platforms: iOS 13.7, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
  IDEs:
    Nano: 2.0.6 - /usr/bin/nano
    VSCode: 1.49.0 - /usr/local/bin/code
    Vim: 8.1 - /usr/bin/vim
    Xcode: 11.7/11E801a - /usr/bin/xcodebuild
  Languages:
    Bash: 3.2.57 - /bin/bash
    Java: javac 15 - /usr/bin/javac
    Perl: 5.18.4 - /usr/bin/perl
    PHP: 7.3.11 - /usr/bin/php
    Python: 2.7.16 - /usr/bin/python
    Python3: 3.8.5 - /usr/local/bin/python3
    Ruby: 2.6.3 - /usr/bin/ruby
    Rust: 1.46.0 - /Users/brianchen/.cargo/bin/rustc
  Databases:
    SQLite: 3.28.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 87.0.4280.141
    Firefox: 83.0
    Safari: 14.0
```

### PR DESCRIPTION
**Pull request by Brianzchen**
Closes #3991.

Basically if there are multiple versions of a library, the tool will first find all lib defs related, it could be
- foo_v1.x.x
- foo_v1.2.x

But if I am looking for foo@1.5.6, it will still pull all of these lib defs run a sort and return me `foo_v1.2.x` as the top matching result, but in fact that one doesn't match so I'm left with the wrong lib def.

In the `filterLibDefs` inside `npmLibDefs.js` it calls `pkgVersionMatch`. This function when it tries to grab `libDefSemver` it's actually done wrong. It only knows to prefix `^`, when it should infact prefix `~` if the flow defined version should only handle on a patch level.

---

- Links to documentation: N/A
- Link to GitHub or NPM: N/A
- Type of contribution: fix

Other notes: delete cli stub defs for rimraf and semver because they already exist with real versions, though they're outdated and should be bumped later

### CODE REVIEW

File: cli/src/lib/npm/npmLibDefs.js [hunk 1]
```diff
@@ -244,13 +244,22 @@ function validateVersionNumPart(part: string, partName: string): number {
...
+  //        and "foo_v2.x.x" is the same range as "^2.x.x"
   // UNLESS it is prefixed by the equals character (i.e. "foo_=v2.2.x")
-  let libDefSemver =
-    libDefSemverRaw[0] !== '=' && libDefSemverRaw[0] !== '^'
-      ? '^' + libDefSemverRaw
-      : libDefSemverRaw;
+  const libDefSemver = (() => {
+    const versionSplit = libDefSemverRaw.split('.');
+    if (libDefSemverRaw[0] !== '=' && libDefSemverRaw[0] !== '^') {
+      if (versionSplit[1] !== 'x') {
```
Comment (Brianzchen) [inline cli/src/lib/npm/npmLibDefs.js:256]: This should handle if versions are `0.4.x` for beta library versions

Comment (Brianzchen) [reply cli/src/lib/npm/npmLibDefs.js:256]: actually should be handled fine https://docs.npmjs.com/cli/v6/using-npm/semver#tilde-ranges-123-12-1

**Comment by Brianzchen** [comment]
@GAntoine thanks for self assigning, it's ready for review now

**Review by GAntoine — APPROVED** [review]

**Comment by GAntoine** [comment]
LGTM. Sorry for taking so long to review these, it's been a very busy month here. I'll release a minor version with all your fixes this week. Thanks a ton for these PRs!

**Comment by Brianzchen** [comment]
Hmm this change added flow errors to the cli. I'll raise a PR to fix them up as soon as I can

**Pull request merged**